CMakeLists.txt

```
@@ -165,6 +165,28 @@ This option is required to support package maintainers that split
     OFF)
 mark_as_advanced(GOOGLE_CLOUD_CPP_USE_INSTALLED_COMMON)
 
+include(CMakeDependentOption)
+cmake_dependent_option(
+    GOOGLE_CLOUD_CPP_WITH_MOCKS
+    [==[Build the google-cloud-cpp mocking libraries.
+
+google-cloud-cpp offers mocking libraries with mock classes, to facilitate unit
+testing of Cloud C++ clients. Consumers of this library that do not use the
+provided mocks to test code involving the Cloud C++ clients may wish to turn
+this flag off.]==]
+    ON
+    "NOT BUILD_TESTING"
+    ON)
+mark_as_advanced(GOOGLE_CLOUD_CPP_WITH_MOCKS)
```

TIL, you can use `NOT condition` in `cmake_dependent_option`.  Using `ON ... ON` looks weird.  I think you want something like this:


```suggestion
option(
    GOOGLE_CLOUD_CPP_WITH_MOCKS
    [==[Build the google-cloud-cpp mocking libraries.

google-cloud-cpp offers mocking libraries with mock classes, to facilitate unit
testing of Cloud C++ clients. Consumers of this library that do not use the
provided mocks to test code involving the Cloud C++ clients may wish to turn
this flag off.]==]
    $<IF:$<BOOL:${BUILD_TESTING}>:OFF:ON>)
mark_as_advanced(GOOGLE_CLOUD_CPP_WITH_MOCKS)
```

That is, the option always exists, but its default value is based on a generator expression.